agora/footfall_ai/src/video_capture.py

```diff
@@ -0,0 +1,38 @@
+import cv2
+import time
+import threading
+import queue
+import cv2, queue, threading, time
+
+# bufferless VideoCapture
+class VideoCapture:
+    def __init__(self, name):
+        self.cap = cv2.VideoCapture(name)
+        self.q = queue.Queue()
+        self.lock = threading.Lock()
+        self.running = True  # Flag to indicate if the thread should keep running
+        self.t = threading.Thread(target=self._reader)
+        self.t.daemon = True
+        self.t.start()
+
+    def _reader(self):
+        while self.running:
+            ret, frame = self.cap.read()
+            if not ret:
+                break
+            if not self.q.empty():
+                try:
+                    self.q.get_nowait()   # Discard previous frame
+                except queue.Empty:
+                    pass
+            
+            self.q.put(cv2.resize(frame, (1280, 720)))
+            #self.q.put(frame)
+            self.state=ret
+
+    def read(self):
+        return self.q.get(),self.state
```

Comment:
We should check the q is not empty here (consumer is much faster than the producer - which will not be the case for this but good to have a check) and handle the exception here with logging and sleeping?

Reply:
@niachary - Agree could you please push your suggested changes?

Reply:
Update on this - self.q.get in python is a blocking call and thus will not throw an error if q is empty and will wait till an item is available in the queue